jenkins/security/whitesource_container_scan.groovy

```diff
@@ -0,0 +1,134 @@
+        string(name: 'CONTROL_EXTERNAL_NODEPORT', defaultValue: '31169', description: 'Port to be used for control service.', trim: true)
+        string(name: 'S3_EXTERNAL_HTTP_NODEPORT', defaultValue: '30080', description: 'HTTP Port to be used for IO service.', trim: true)
+        string(name: 'S3_EXTERNAL_HTTPS_NODEPORT', defaultValue: '30443', description: 'HTTPS to be used for IO service.', trim: true)
+        text(defaultValue: '''hostname=<hostname>,user=<user>,pass=<password>''', description: 'VM details to be used. First node will be used as Primary node', name: 'hosts')
+        choice(
+            name: 'EXTERNAL_EXPOSURE_SERVICE',
+            choices: ['LoadBalancer', 'NodePort'],
+            description: 'K8s Service to be used to expose RGW Service to outside cluster.'
+        )
+        string(name: 'WHITESOURCE_SERVER_URL', defaultValue: 'https://saas.whitesourcesoftware.com', description: 'WhiteSource Server URL', trim: true)
+        string(name: 'API_KEY', defaultValue: ' ', description: 'apiKey', trim: true)
+        string(name: 'USER_KEY', defaultValue: ' ', description: 'userKey', trim: true)
+        string(name: 'PRODUCT_NAME', defaultValue: 'cortx-all-k8s-cluster', description: 'productName', trim: true)
+        string(name: 'DOCKER_REGISTRY', defaultValue: 'ghcr.io/seagate', description: 'docker registry', trim: true)
+        string(name: 'WHITESOURCE_VERSION', defaultValue: '20.11.1', description: 'mainPod & workerPod version', trim: true)
+        string(name: 'PULL_SECRET', defaultValue: ' ', description: 'image pullSecretL', trim: true)
+        // Please configure CORTX_SCRIPTS_BRANCH and CORTX_SCRIPTS_REPO parameter in Jenkins job configuration.
+    }    
+
+    stages {
+
+        stage('Checkout Script') {
+            steps { 
+                cleanWs()            
+                script {
+                    checkout([$class: 'GitSCM', branches: [[name: "${CORTX_RE_BRANCH}"]], doGenerateSubmoduleConfigurations: false, extensions: [], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'cortx-admin-github', url: "${CORTX_RE_REPO}"]]])                
+                }
+            }
+        }
+
+        stage ('Setup K8s Cluster') {
+            steps {
+                script { build_stage = env.STAGE_NAME }
+                sh label: 'Setup K8s Cluster', script: '''
+                    pushd solutions/kubernetes/
+                        echo $hosts | tr ' ' '\n' > hosts
+                        cat hosts
+                        if [ "$(cat hosts | wc -l)" -eq 2 ]
+                        then
+                           echo "Current configuration does not support 2 node CORTX cluster deployment. Please try with 1 or more than two nodes."
+                           echo "Exiting Jenkins job."
+                           exit 1
+                        fi
+                        ./cluster-setup.sh ${PODS_ON_PRIMARY}
```

Comment:
Do we need `${PODS_ON_PRIMARY}` if we are always planing to use 3+1 setup

Reply:
I have seen your previous comment, as we are yet to receive the update from swanand so i didn't took any action on it. if PODS_ON_PRIMARY = "false" do we still need to declare it explicitly? @shailesh-vaidya

Reply:
We do not need to declare it if it's false. Please follow up with @swanand-gadre and get it confirmed.

Reply:
removed ${PODS_ON_PRIMARY}